### ISSUE TITLE
Hot update not working

### ISSUE THREAD

**Issue by zxcabs** [webpack/style-loader#53]
Hello,
I will try to use hot update for style-loader with file and less loaders and it doesn't work.
Here example https://github.com/zxcabs/webpack-hot-update-example
When i changed .less file webpack recompiles the build, but style doesn't changed.

what i do wrong?

**Comment by sokra** [comment]
You are not using the style-loader: https://github.com/zxcabs/webpack-hot-update-example/blob/master/webpack.config.js#L25

**Comment by zxcabs** [comment]
I'm included it in source https://github.com/zxcabs/webpack-hot-update-example/blob/master/src/index.js#L5

it's wrong?

**Comment by jhnns** [comment]
@sokra I've created a minimal example to illustrate the problem based on @zxcabs 's example: https://github.com/jhnns/webpack-hot-update-example

It looks like the module request is translated into a module id two times:
![](https://cloud.githubusercontent.com/assets/781746/9088179/0a04cc96-3b8f-11e5-8f2d-1b48ee0e69a0.jpg)
`5959` should be just `59`

**Comment by jhnns** [comment]
When I hard-code `59` into this line, a Maximum call-stack error is thrown.

**Comment by matt-d-rat** [comment]
I'm experiencing a similar/the same problem. I spent ages trying to figure out my seed project (React + ES6 + babel + css-loader (__with [css-modules](https://github.com/webpack/css-loader#css-modules) feature turned on__) ) wasn't working. 

I had to resort to using the extract-text-webpack-plugin to work around it. But this still doesn't give me HMR for styles.

Essentially when I wasn't using extract-text-webpack-plugin and my webpack.config.js is:

__webpack.config.js__
```
'use strict';

var webpack = require('webpack');

module.exports = {
    output: {
        filename: 'main.js',
        publicPath: '/assets/'
    },

    cache: true,
    debug: true,
    devtool: 'sourcemap',
    entry: [
        'webpack/hot/only-dev-server',
        './src/components/main.js'
    ],

    stats: {
        colors: true,
        reasons: true
    },

    resolve: {
        modulesDirectories: ['node_modules', 'components'],
        extensions: ['', '.js', '.jsx'],
        alias: {
            'styles': __dirname + '/src/styles',
            'mixins': __dirname + '/src/mixins',
            'components': __dirname + '/src/components/',
            'stores': __dirname + '/src/stores/',
            'actions': __dirname + '/src/actions/'
        }
    },
    module: {
        preLoaders: [
            {
                test: /\.(js|jsx)$/,
                exclude: /node_modules/,
                loader: 'eslint-loader'
            }
        ],
        loaders: [{
            test: /\.(js|jsx)$/,
                exclude: /node_modules/,
                loader: 'react-hot!babel-loader'
            },
            {
                test: /\.css$/,
                loader: 'style-loader!css-loader?modules&importLoaders=1&localIdentName=[name]__[local]___[hash:base64:5]!postcss-loader'
            },
            {
                test: /\.scss/,
                loader: 'style-loader!css-loader?modules&importLoaders=2&localIdentName=[name]__[local]___[hash:base64:5]!postcss-loader!sass-loader?outputStyle=expanded'
            },
            {
                test: /\.(png|jpg|woff|woff2)$/,
                loader: 'url-loader?limit=8192'
            },
            {
                test: /\.svg$/, 
                loader: 'url-loader?limit=10000&mimetype=image/svg+xml'
            }
        ]
    },

    postcss: [
        require('autoprefixer')
    ],

    plugins: [
        new webpack.HotModuleReplacementPlugin()
    ]
};

```

When I imported or required the CSS file:

```
import styles from './App.scss';
```

```
var styles = require('./App.scss');
```

```styles``` would be an empty object. If I removed style-loader from the __webpack.config.js__ then this variable would be populated.

**Comment by tomchentw** [comment]
I experienced the same issue as today. Use all `@latest` package from npm and the error still exists.

## The error is exactly what @jhnns described

The first argument of `module.hot.accept` function call inside `if(!content.locals) {` block will evaluate to `7777` on my computer, which is obviously no such module. It should be just `77`.

However, in this case, it won't throw any error and people think they're working fine because `7777` is still a valid JS syntax. I've found a situation that will cause syntax error to the source.

## Add `output.pathinfo: true` in webpack.config.js

I've forked the project from @jhnns and you can see I only had one commit difference, which adds `output.pathinfo: true` to webpack.config.js. You could try it here:
https://github.com/tomchentw/webpack-hot-update-example

Here's the screenshot when I visited the browser:

<img width="885" alt="screen shot 2016-01-20 at 10 13 05 am" src="https://cloud.githubusercontent.com/assets/922234/12438192/2e714fe2-bf5f-11e5-85e9-4be06181be3a.png">

And here's the log in pure text:

```js
/***/ },
/* 76 */
/*!***********************!*\
  !*** ./src/style.css ***!
  \***********************/
/***/ function(module, exports, __webpack_require__) {

  // style-loader: Adds some css to the DOM by adding a <style> tag

  // load the styles
  var content = __webpack_require__(/*! !./../~/css-loader!./style.css */ 77);
  if(typeof content === 'string') content = [[module.id, content, '']];
  // add the styles to the DOM
  var update = __webpack_require__(/*! ./../~/style-loader/addStyles.js */ 79)(content, {});
  if(content.locals) module.exports = content.locals;
  // Hot Module Replacement
  if(true) {
    // When the styles change, update the <style> tags
    if(!content.locals) {
      module.hot.accept(/*! !./../~/css-loader!./style.css */ 77/*! !./../~/css-loader!./style.css */ 77, function() {
        var newContent = __webpack_require__(/*! !./../~/css-loader!./style.css */ 77);
        if(typeof newContent === 'string') newContent = [[module.id, newContent, '']];
        update(newContent);
      });
    }
    // When the module is disposed, remove the <style> tags
    module.hot.dispose(function() { update(); });
  }

/***/ },
/* 77 */
/*!**************************************!*\
  !*** ./~/css-loader!./src/style.css ***!
  \**************************************/
```

The `/*! !./../~/css-loader!./style.css */ 77/*! !./../~/css-loader!./style.css */ 77` won't be a valid JS syntax and an exception was thrown.

**Comment by tomchentw** [comment]
I had setup an example that use JavaScript as `entry` and require css in that entry file. However, the result is the same. Branch is here:
https://github.com/tomchentw/webpack-hot-update-example/compare/master...tomchentw:feat/use-js-entry?expand=1

**Comment by jhnns** [comment]
Ok, I found the problem.

The issue occurs when the `HotModuleReplacementPlugin` is applied twice. In my example, I used the `webpack-dev-server` with the `--hot` option *and* applied the plugin in the `webpack.config.js`. There might be other cases as well where the HotModuleReplacementPlugin is already applied.

I just pushed a fix to my example repository. After that, the HMR works as expected.

Close @zxcabs?
